themes/default/content/blog/top-5-things-for-azure-devs-vm/index.md

```diff
@@ -0,0 +1,231 @@
+meta_image: azure-top-5.png
+authors:
+    - sophia-parafina
+tags:
+    - azure
+    - virtual machines
+---
+
+So you want to be an Azure dev and all-around infrastructure wizard? Let's start with the basics, virtual machines! In the previous article, the common use case for virtual machines is migrating applications from dedicated hardware. You want full control of the machine to install required software for the application or configure storage and networking.
+
+Azure provides many ways to create and configure virtual machines ranging from the Azure CLI to the Azure Portal. In this article, we'll first create a virtual machine using the portal to understand the requirements and process; then, we'll do it with code using Pulumi's Azure Native provider, which is built directly from the Azure API.
+
+A native provider provides functionality mapped directly from the underlying API; in the case of the new Azure Provider, functionality is mapped directly from the Azure Resource Manager API surface.
+
+<!--more-->
+
+## Creating a VM with the Azure Portal
+
+When creating virtual machines, you should consider the operating system, storage requirements, and vm size regarding CPU to memory ratio. Azure offers [many sizes for virtual machines](https://docs.microsoft.com/en-us/azure/virtual-machines/sizes) depending on your workload. If you're migrating a traditional n-tier application to the cloud, it's best to start with virtual machines that match your current configuration.
+
+This article will use a simple example of deploying a static web application on a virtual machine to illustrate the process step by step. To get started, sign in to the [Azure Portal](https://portal.azure.com/).
+
+**Step 1**:
+
+Under `Azure Services`, select `Virtual machines`
+
+![Azure portal](./image3.png)
+
+Select `Create` > `Virtual Machine`
+
+![Create VM](./image8.png)
+
+**Step 2**: Create a Resource Group
+
+A Resource Group is a logical collection of virtual machines, storage, virtual networks, web apps, and databases. Users will commonly group an application's related resources. We'll create `myResourceGroup` for our application.
+
+![Resource group](./image6.png)
+
+**Step 3**: Configure the VM
+
+Next, we'll provide details for the vm by giving it a name, `myVM`, setting the `Region` and `Availability options` to `No infrastructure redundancy required`. We'll use an Ubuntu image and set the vm size `Standard_DS1_v2`.
+
+![Configure VM](./image2.png)
+
+**Step 4**: Configure the Administrator Account
+
+Virtual machines are Infrastructure as a Service, which means you will have to perform provisioning, updates, and maintenance. You'll need an administrator account to perform these tasks. To keep this example simple, we'll use `Password` as the `Authentication type`, but in practice, you should use Public Key Infrastructure (PKI) because it is more secure than a password that can be brute-forced.
```

Comment:
```suggestion
Virtual machines are Infrastructure as a Service, which means you will have to perform provisioning, updates, and maintenance. You'll need an administrator account to perform these tasks. To keep this example simple, we'll use `Password` as the `Authentication type`, but in practice, you should use [Public Key Infrastructure (PKI)](https://en.wikipedia.org/wiki/Public_key_infrastructure) because it is more secure than a password that can be brute-forced.
```